src/widget/wfindonwebmenu.cpp

```diff
@@ -0,0 +1,269 @@
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackSearchProperty, albumName]() {
+                    emit triggerBrowser(service, trackSearchProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackSearchProperty, albumName]() {
+                    emit triggerBrowser(service, trackSearchProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackSearchProperty, albumName]() {
+                    emit triggerBrowser(service, trackSearchProperty, albumName);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnWebMenu::addActionsTrackTitle(Service service,
+        const QString& trackTitle,
+        QMenu* pServiceMenu,
+        TrackSearchProperties trackSearchProperty) {
+    const QString prefixAction = composePrefixAction(trackSearchProperty);
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackSearchProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackSearchProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackSearchProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackSearchProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackSearchProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackSearchProperty, trackTitle);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnWebMenu::openInBrowser(Service service,
+        TrackSearchProperties trackSearchProperty,
+        const QString& query) {
+    const QString serviceUrl = getServiceUrl(service, trackSearchProperty);
+    QUrlQuery urlQuery;
+    urlQuery.addQueryItem("q", query);
+    QUrl url(serviceUrl);
+    url.setQuery(urlQuery);
+    QDesktopServices::openUrl(url);
+}
+
+void WFindOnWebMenu::populateFromTrackProperties(
+        const Track& track,
+        const QString& serviceTitle,
+        Service service) {
+    const auto artist = track.getArtist();
+    m_pServiceMenu = new QMenu(this);
```

Comment:
This could be memory leak if `m_pServiceMenu` is re-initialized while not being owned by someone else.

Why a member variable? I don't see that it is used anywhere?